src/Pages/OverviewPages/SolarOnboarding.js

```diff
@@ -0,0 +1,624 @@
+                { title: 'Name', dataIndex: 'name', key: 'name' },
+                { title: 'Branch', dataIndex: 'branch_name', key: 'branch_name' },
+                { title: 'Installed Capacity (kWp)', dataIndex: 'installed_capacity', key: 'installed_capacity' },
+                {
+                  title: 'Active',
+                  dataIndex: 'is_active',
+                  key: 'is_active',
+                  render: (val) => (
+                    <span style={{ color: val ? 'green' : 'red', fontWeight: 'bold' }}>{val ? 'Yes' : 'No'}</span>
+                  ),
+                },
+                { title: 'Address', dataIndex: 'address', key: 'address' },
+                { title: 'Latitude', dataIndex: 'latitude', key: 'latitude' },
+                { title: 'Longitude', dataIndex: 'longitude', key: 'longitude' },
+                { title: 'Created At', dataIndex: 'created_at', key: 'created_at' },
+                {
+                  title: 'Action',
+                  key: 'action',
+                  render: (_, record) => {
+                    const loadingKey = record.id ?? record.branch_id ?? record.branch;
+                    const isToggling = toggleStationStatusLoadingId === loadingKey;
+                    const isActive = record.is_active;
+                    return (
+                      <Button
+                        variant="solid"
+                        color={isActive ? 'danger' : 'primary'}
+                        danger={isActive}
+                        loading={isToggling}
+                        onClick={(event) => {
+                          event.stopPropagation();
+                          handleToggleStationStatus(record);
+                        }}
+                        disabled={isToggling}
+                      >
+                        {isActive ? 'Deactivate' : 'Activate'}
+                      </Button>
+                    );
+                  },
+                },
+              ]}
+              rowKey={(record) => record.id || record.deye_station_id}
+              loading={stationsLoading}
+              pagination={{ pageSize: 10 }}
+              onRow={(record) => {
+                const branchIdentifier = record?.branch_id ?? record?.branch;
+                return {
+                  onClick: () => {
+                    if (!branchIdentifier) {
+                      notification.warning({
+                        message: 'Branch Missing',
+                        description: 'Branch information is required to view station details.',
+                      });
+                      return;
+                    }
+                    navigate(`/solar-onboarding/stations/${branchIdentifier}`, {
+                      state: {
+                        stationId: record?.id,
+                        station: record,
+                      },
+                    });
+                  },
```

Comment:
@DevIsraelHub , I will be merging this, but lets try to move things in onclick to a separate function